pkg/router/route_group.go

```diff
@@ -219,20 +226,14 @@ func (rg *RouteGroup) Write(p []byte) (n int, err error) {
 	}
 }
 
-func (rg *RouteGroup) writePacketAsync(tp *transport.ManagedTransport, packet routing.Packet) (chan error, func()) {
-	ctx, cancel := context.WithCancel(context.Background())
-
+func (rg *RouteGroup) writePacketAsync(ctx context.Context, tp *transport.ManagedTransport, packet routing.Packet) chan error {
 	errCh := make(chan error)
```

Comment:
Should we initialize this with size of 1 so the `WritePacket` go routine does not hang?

Reply:
@evanlinjin I'll think a bit

Reply:
@evanlinjin the only reason for having a err chan here is to be able to do a select between it and the context chan. so, the err chan gets contantly listened to outside of `writePacketAsync` and this way we really want to block on this